### PR TITLE
Add mapping for multiple environments

### PR DESCRIPTION
**Pull request by marySalvi**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Replace `# (issue)` below with a reference to the issue that this PR addresses, of the form `#\d+`. If merging this PR should close the issue, use `Fixes` or `Closes`. Then delete this paragraph.

Fixes # [1367](https://github.com/microbiomedata/nmdc-server/issues/1367)

## Type of change

Please delete options that are not relevant.

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update  _I'm not sure_

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration, if it is not simply `make up-test && make test-run`.

- [x] Adjusted test fakes to represent multiple environments 

**Configuration Details**: none

# Definition of Done (DoD) Checklist:
A simple checklist of necessary activities that add verifiable/demonstrable value to the product by asserting the quality of a feature, not the functionality of that feature; the latter should be stated as acceptance criteria in the issue this PR closes. Not all activities in the PR template will be applicable to each feature since the definition of done is intended to be a comprehensive checklist. Consciously decide the applicability of value-added activities on a feature-by-feature basis.

- [x] My code follows the style guidelines of this project (have you run `black nmdc_runtime/`?)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (in `docs/` and in <https://github.com/microbiomedata/NMDC_documentation/>?)
- [ ] I have added tests that prove my fix is effective or that my feature works, incl. considering downstream usage (e.g. <https://github.com/microbiomedata/notebook_hackathons>) if applicable.
- [x] New and existing unit and functional tests pass locally with my changes (`make up-test && make test-run`)

### CODE REVIEW

**Comment by marySalvi** [comment]
@mslarae13  We are meeting tomorrow to debug and get a working solution. I will post an updated ETA as soon as possible.

File: nmdc_runtime/site/translation/submission_portal_translator.py [hunk 1]
```diff
@@ -1,21 +1,44 @@
...
+    BIOFILM = "microbial mat_biofilm"
+    BUILT_ENV = "built environment"
+    HCR_CORES = "hydrocardbon resources-cores"
+    HRC_FLUID_SWABS = "hydrocarbon resources-fluids_swabs"
+    HOST_ASSOCIATED = "host-associated"
+    MISC_ENVS = "miscellaneous natural or artificial environment"
+    PLANT_ASSOCIATED = "plant-associated"
+    SEDIMENT = "sediment"
+    SOIL = "soil"
+    WATER = "water"
```
Comment (dwinston) [inline nmdc_runtime/site/translation/submission_portal_translator.py:39]: issue here related to https://github.com/microbiomedata/submission-schema/issues/186

File: nmdc_runtime/site/translation/submission_portal_translator.py [hunk 2]
```diff
@@ -554,17 +576,17 @@ def _translate_biosample(
...
         for tab in sample_data:
+            biosample_key = tab.get(BIOSAMPLE_UNIQUE_KEY_SLOT, "").strip()
+            slots = {
+                "id": nmdc_biosample_id,
+                "associated_studies": [nmdc_study_id],
+                "type": "nmdc:Biosample",
+                "name": tab.get("samp_name", "").strip(),
+                "env_package": nmdc.TextValue(
+                    has_raw_value=tab.get("env_package"), type="nmdc:TextValue"
+                ),
```
Comment (pkalita-lbl) [inline nmdc_runtime/site/translation/submission_portal_translator.py:588]: By moving this definition to inside the loop, won't this only build a `Biosample` based on the _last_ tab it sees? So for example, metadata about one sample can be contained in both the `soil_data` tab and the `emsl_data` tab. Wouldn't this code first process the information from the soil tab and then completely replace it with information from the EMSL tab?

**Review by pkalita-lbl — CHANGES_REQUESTED** [review]
I think a lot of this is on the right track, but it seems to be missing a few pieces related to how the JGI and EMSL tabs are handled in conjunction with the multiple environments. If you want to go over this in more detail together just let me know.

File: nmdc_runtime/site/translation/submission_portal_translator.py (hunk 2 above)
Comment (marySalvi) [reply nmdc_runtime/site/translation/submission_portal_translator.py:588]: Yes, I was thinking weren't using the mixin templates but rather just the environments. I will change this.

File: nmdc_runtime/site/translation/submission_portal_translator.py [hunk 3]
```diff
@@ -1,21 +1,44 @@
...
+    >>> EnvironmentType.AIR.value
+    'air'
+    >>> EnvironmentType.SEDIMENT.value
+    'sediment'
+    """
+
+    AIR = "air"
+    BIOFILM = "microbial mat_biofilm"
+    BUILT_ENV = "built environment"
+    HCR_CORES = "hydrocardbon resources-cores"
```
[the remark this answers is not in the document]
Comment (marySalvi) [reply nmdc_runtime/site/translation/submission_portal_translator.py]: fixed in 49456dade6a9521eb09a9bfd128886fb0881cc66

File: nmdc_runtime/site/translation/submission_portal_translator.py [hunk 4]
```diff
@@ -599,10 +621,16 @@ def get_database(self) -> nmdc.Database:
         ]
 
         sample_data = metadata_submission_data.get("sampleData", {})
-        package_name = metadata_submission_data["packageName"]
-        sample_data_by_id = groupby(
-            BIOSAMPLE_UNIQUE_KEY_SLOT, concat(sample_data.values())
-        )
+        for key in sample_data.keys():
+            env = key.rsplit("_", 1)[0].upper()
```
[the remark this answers is not in the document]
Comment (marySalvi) [reply nmdc_runtime/site/translation/submission_portal_translator.py]: changed in 49456dade6a9521eb09a9bfd128886fb0881cc66

File: nmdc_runtime/site/translation/submission_portal_translator.py [hunk 5]
```diff
@@ -599,10 +621,16 @@ def get_database(self) -> nmdc.Database:
...
+            package_name = EnvironmentType[env].value
+
+            sample_data_by_id = groupby(
+                BIOSAMPLE_UNIQUE_KEY_SLOT,
+                concat(sample_data.values()),
+            )
```
[the remark this answers is not in the document]
Comment (marySalvi) [reply nmdc_runtime/site/translation/submission_portal_translator.py]: fixed in 49456dade6a9521eb09a9bfd128886fb0881cc66

File: nmdc_runtime/site/translation/submission_portal_translator.py (hunk 2 above)
Comment (marySalvi) [reply nmdc_runtime/site/translation/submission_portal_translator.py]: logic adjusted in 38a14c50717ec3cc6bfc97928dfeb17531262519

**Comment by pkalita-lbl** [comment]
@marySalvi Sorry I sort of led you astray a bit with my initial feedback. After looking at this again the real key is not adding the analysis templates (the EMSL and JGI ones) into the enum, but rather to _skip_ setting the `env_package` slot while traversing the keys containing that data. Then while processing an individual sample's data, the logic is to find the first (in reality only) object where `env_package` is set to identify the environmental data.

As to not have a bunch more back and forth on this I went ahead and expanded the test data to capture this use case and updated the translator logic.

**Review by pkalita-lbl — APPROVED** [review]

**Pull request merged**